x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/ExecuteStepsUpdateTask.java

```diff
@@ -86,12 +86,11 @@ public ClusterState execute(final ClusterState currentState) throws IOException
             // either get to a step that isn't a cluster state step or a
             // cluster state wait step returns not completed
             while (currentStep instanceof ClusterStateActionStep || currentStep instanceof ClusterStateWaitStep) {
-                nextStepKey = currentStep.getNextStepKey();
                 if (currentStep instanceof ClusterStateActionStep) {
                     // cluster state action step so do the action and
                     // move the cluster state to the next step
-                    logger.trace("[{}] performing cluster state action ({}) [{}], next: [{}]",
-                        index.getName(), currentStep.getClass().getSimpleName(), currentStep.getKey(), currentStep.getNextStepKey());
+                    logger.trace("[{}] performing cluster state action ({}) [{}]",
```

Comment:
Maybe since we're losing the trace logging for where we're going to move to on the next step, we can add it below before we call `moveClusterStateToNextStep`?

Reply:
++